### PR TITLE
Add recommendationsLock

### PR DESCRIPTION
**Pull request by jacky68147527**
Add recommendationsLock to solve when we start multiple workers, goroutine read and write recommendations may unsafe.

### CODE REVIEW

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/bk-bcs?pullRequest=1904) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/bk-bcs?pullRequest=1904) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: jacky68147527<br/>:x: hansenwu<br/><hr/>**hansenwu** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/bk-bcs?pullRequest=1904) it.</sub>

File: bcs-runtime/bcs-k8s/bcs-component/bcs-general-pod-autoscaler/pkg/scaler/general.go [hunk 1]
```diff
@@ -99,6 +99,9 @@ type GeneralController struct {
 	// Latest unstabilized recommendations for each autoscaler.
 	recommendations map[string][]timestampedRecommendation
 
+	// Multi goroutine read and write recommendations may unsafe.
+	recommendationsLock sync.Mutex
+
 	// Latest autoscaler events
 	scaleUpEvents   map[string][]timestampedScaleEvent
 	scaleDownEvents map[string][]timestampedScaleEvent
```
Comment (xcming) [inline bcs-runtime/bcs-k8s/bcs-component/bcs-general-pod-autoscaler/pkg/scaler/general.go:107]: up and down events also need to be locked.

File: bcs-runtime/bcs-k8s/bcs-component/bcs-general-pod-autoscaler/pkg/scaler/general.go [hunk 2]
```diff
@@ -919,6 +922,10 @@ func (a *GeneralController) computeStatusForExternalMetric(
 }
 
 func (a *GeneralController) recordInitialRecommendation(currentReplicas int32, key string) {
+	// add lock
+	a.recommendationsLock.Lock()
+	defer a.recommendationsLock.Unlock()
+
```
Comment (xcming) [inline bcs-runtime/bcs-k8s/bcs-component/bcs-general-pod-autoscaler/pkg/scaler/general.go:928]: 4 functions (reconcileKey, storeScaleEvent, stabilizeRecommendationWB, convertDesiredReplicasWithBR) should add locks too.

Comment (jacky68147527) [reply bcs-runtime/bcs-k8s/bcs-component/bcs-general-pod-autoscaler/pkg/scaler/general.go:946]: Thanks for your help, scaleUp/scaleDown events lock has been added.

**Review by xcming — APPROVED** [review]

**Comment by wenxinlee2015** [comment]
@jacky68147527 Please deal with the GPG issue

**Pull request merged**